diff -u test.zig test.zig
--- test.zig	2025-02-19 16:35:35.052229398 +0100
+++ test.zig	2025-02-19 16:35:35.054076780 +0100
@@ -1161,7 +1161,8 @@
     defer tmp.cleanup();
 
     try tmp.dir.makeDir("A");
-    const tmpA = try tmp.dir.openDir("A", .{});
+    var tmpA = try tmp.dir.openDir("A", .{});
+    defer tmpA.close();
     try tmpA.makeDir("B");
 
     const testPath = "A" ++ fs.path.sep_str ++ "B" ++ fs.path.sep_str ++ "C";
